unitconfig/unitconfig.go

```diff
@@ -176,14 +168,24 @@ func (instance *Instance) load() (err error) {
 	return nil
 }
 
-func (instance *Instance) checkUnitConfig(unitConfig cloudprotocol.UnitConfig) (vendorVersion string, err error) {
-	if unitConfig.Version == instance.unitConfig.Version {
-		return unitConfig.Version, ErrAlreadyInstalled
+func (instance *Instance) checkVersion(version string) error {
+	curVer, err := semver.NewVersion(instance.unitConfig.Version)
+	if err != nil {
+		return aoserrors.Wrap(err)
 	}
 
-	if err = instance.client.CheckUnitConfig(unitConfig); err != nil {
-		return unitConfig.Version, aoserrors.Wrap(err)
+	newVer, err := semver.NewVersion(version)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	if newVer.Equal(curVer) {
+		return ErrAlreadyInstalled
 	}
 
-	return unitConfig.Version, nil
+	if newVer.LessThan(curVer) {
+		return aoserrors.New("wrong version")
```

Comment:
Optional,
consider returning more specific error, for example "version downgrade is prohibited"

Reply:
Will keep as is.